### PR TITLE
loading search, diamonds added

### PR DESCRIPTION
**Pull request by barakavraham**

### CODE REVIEW

File: app/api/base.py [hunk 1]
```diff
@@ -18,6 +18,13 @@ def search_for_resources(self):
         resourceFound = math.ceil(current_user.army.field/randomPercentage)
         return resourceFound
 
+    def search_for_diamond(self):
+        randomNum = random.randint(0,10)
+        if randomNum == 1:
```
Comment (daniel-atbay) [inline app/api/base.py]:
```suggestion
        if randomNum == 1:
        return 1 if random_num == 1 else 0
```

File: app/api/base.py [hunk 2]
```diff
@@ -34,6 +41,10 @@ def add_user_resources(self):
             'coin': {
                 'amount': self.search_for_resources(),
                 'picture': url_for ('static', filename=f'images/coin.png')
+            },
+            'diamond': {
+                'amount': self.search_for_diamond(),
+                'picture': url_for ('static', filename=f'images/diamond.png')
```
Comment (daniel-atbay) [inline app/api/base.py]: Why are these f strings?
```suggestion
                'picture': url_for ('static', filename='images/diamond.png')
```

File: app/api/base.py [hunk 3]
```diff
@@ -54,3 +65,39 @@ def get(self):
...
+    def can_search(self):
+        return current_user.army.turns >= 15
+
+    def search_for_resources(self):
+        random_field = random.randint(50,100)
+        return random_field
+
+    def add_user_resources(self):
+        add_resource = {
+            'field': {'amount': self.search_for_resources(), 'picture': url_for ('static', filename=f'images/field.png') }
```
Comment (daniel-atbay) [inline app/api/base.py]:
```suggestion
            'field': {
                'amount': self.search_for_resources(),
                'picture': url_for ('static', filename=f'images/field.png')
            }
```

File: app/static/js/base.js [hunk 4]
```diff
@@ -5,45 +5,113 @@
...
                 $card_body.find('.col').last().append(`<p class="card-text mb-0"><img class="resource-img img-fluid" src="${added_resource[resource].picture}" /></p>`);
                 $card_body.find('.col').last().append(`<p class="card-text">${added_resource[resource].amount}</p>`);
             }
         }
 
+        function setPurchaseResultsMessage($purchaseResultsDiv, purchaseSuccess, text) {
+            $purchaseResultsDiv.removeClass('message-showing').empty();
+            $purchaseResultsDiv.append(`<p>${text}</p>`).addClass('message-showing');
+            if (purchaseSuccess)
+                $purchaseResultsDiv.find('p').addClass('bg-success');
```
Comment (daniel-atbay) [inline app/static/js/base.js]: For identical lines with different values you can do this.
This checks if purchaseSuccess is true. If so, we use 'bg-success', else, we use 'bg-warning'
On google, search javascript ternary operator
```suggestion
                $purchaseResultsDiv.find('p').addClass(purchaseSuccess ? 'bg-success' : 'bg-warning');
```

File: app/static/js/base.js [hunk 5]
```diff
@@ -5,45 +5,113 @@
...
 
                 $.gameApiGet('base/search_resources').done(function(user_new_resources){
-                    setUserResources($btn, user_new_resources, $userResources);
-                    setCardBody($btn, user_new_resources);
+                    setSpinner($btn, user_new_resources);
+                    setTimeout(function(){setUserResources(user_new_resources, $userResources)}, 3000);
+                    setTimeout(function(){setCardBody($btn, user_new_resources)}, 3000);
+                }).fail(function(status){
+                    if (status === 400)
+                        setPurchaseResultsMessage($purchaseResult, purchaseSuccess, "You don't have enough turns");
```
Comment (daniel-atbay) [inline app/static/js/base.js]:
```suggestion
                        setPurchaseResultsMessage($purchaseResult, purchaseSuccess, status == 400 ? "You don't have enough turns" : 'Status code');
```

File: app/static/js/base.js [hunk 6]
```diff
@@ -5,45 +5,113 @@
...
                     return false;
+                }
 
                 $.gameApiGet('base/search_resources').done(function(user_new_resources){
-                    setUserResources($btn, user_new_resources, $userResources);
-                    setCardBody($btn, user_new_resources);
+                    setSpinner($btn, user_new_resources);
+                    setTimeout(function(){setUserResources(user_new_resources, $userResources)}, 3000);
+                    setTimeout(function(){setCardBody($btn, user_new_resources)}, 3000);
+                }).fail(function(status){
```
Comment (daniel-atbay) [inline app/static/js/base.js]: Did you mean this?
```suggestion
                }).fail(function({ status }){
```

File: app/static/js/base.js [hunk 7]
```diff
@@ -5,45 +5,113 @@
...
+                    if (status === 400)
+                        setPurchaseResultsMessage($purchaseResult, purchaseSuccess, "You don't have enough turns");
+                    else
+                        setPurchaseResultsMessage($purchaseResult, purchaseSuccess, "Status code");
+                })
+            });
+        }
+
+        function setupFieldSearch() {
+            $('#search-field-btn').on('click', function(){
```
Comment (daniel-atbay) [inline app/static/js/base.js]: Try adding spaces between function declarations and `{`, looks cleaner
```suggestion
            $('#search-field-btn').on('click', function() {
```

File: migrations/versions/fbdd3a81be2a_lol.py [hunk 8]
```diff
@@ -1,16 +1,16 @@
-"""init db
+"""lol
```
Comment (daniel-atbay) [inline migrations/versions/fbdd3a81be2a_lol.py]: wut

**Review by daniel-atbay — APPROVED** [review]

File: app/api/base.py (hunk 2 above)
Comment (barakavraham) [reply app/api/base.py]: my bad, copy paste :(

Comment (daniel-atbay) [reply app/api/base.py]: You don't need it in the other ones either

File: app/static/js/base.js (hunk 4 above)
Comment (barakavraham) [reply app/static/js/base.js]: this is pretty nice, thanks

File: app/static/js/base.js (hunk 7 above)
Comment (barakavraham) [reply app/static/js/base.js]: alright

File: migrations/versions/fbdd3a81be2a_lol.py (hunk 8 above)
Comment (barakavraham) [reply migrations/versions/fbdd3a81be2a_lol.py]: lol

File: app/api/base.py (hunk 2 above)
Comment (barakavraham) [reply app/api/base.py]: Yea i saw, thanks

**Pull request merged**